rakefile

```diff
@@ -0,0 +1,89 @@
+require 'rototiller'
+
+task :default => :acceptance
+
+desc 'Run Performance setup'
+rototiller_task :performance do |t|
+  t.add_env({:name => 'ENVIRONMENT_TYPE', :message => 'Either gatling or clamps', :default => 'gatling'})
+
+  if ENV['ENVIRONMENT_TYPE'] == 'clamps'
+    default_tests = 'setup/install'
+  else
+    default_tests = 'setup/install/00_pre_install/,setup/install/10_install/00_install_pe.rb,setup/install/30_post_install/'
+  end
+
+  # env vars needed for Beaker
+  t.add_command do |command|
+    command.name = "bundle exec beaker"
+    command.add_env({:name => 'BEAKER_EXECUTABLE'})
+
+    command.add_option do |option|
+      option.name = '--config'
+      option.message = 'The configuration file that Beaker will use'
+      option.add_argument do |arg|
+        arg.add_env({:name => 'CONFIG'})
+      end
+    end
+    command.add_option do |option|
+      option.name = '--keyfile'
+      option.message = 'The SSH key used to access a SUT'
+      option.add_argument do |arg|
+        arg.name = "#{ENV['HOME']}/.ssh/id_rsa-acceptance"
+        arg.add_env({:name => 'BEAKER_KEYFILE'})
+      end
+    end
+    command.add_option do |option|
+      option.name = '--log-level'
+      option.message = 'The log level under which you want beaker to run'
+      option.add_argument do |arg|
+        arg.name = 'debug'
+        arg.add_env({:name => 'BEAKER_LOG_LEVEL'})
+      end
+    end
+    command.add_option do |option|
+      option.name = '--pre-suite'
+      option.message = 'Beaker pre-suite'
+      option.add_argument do |arg|
+        arg.name = default_tests
+        arg.add_env({:name => 'BEAKER_PRE_SUITE'})
+      end
+    end unless ENV['BEAKER_PRE_SUITE'] == ''
```

Comment:
end unless?  What does that do?

Reply:
The unless applies to the entire block inside do/end.  This makes the parameter optional.

Reply:
it's like do/while in perl.  first iteration in loop is non-optional

Reply:
Sam explained it all to me.  Do/while is evil though. :)  The decision logic ideally should be at the top.  Because people read top down.  But that is just my opinion.  :)

Reply:
Yeah, it's probably easier to parse in single line statements.  I moved this to the top.

Reply:
agreed.  one can often put the decision logic at the top so it's a bit easier to read and then use/update that variable inside the loop.  the same thing happens when you have a non-deterministic loop count that needs a break in the middle.  especially in a language that doesn't have a for(x;y;z) type syntax that makes the break easy to see up top <cough> ruby.  of course, that's because in ruby you are usually iterating over an array or hash